File: src/lib/cv/contact-photo.ts
import type { ContactProfile } from "@/lib/types/cv";

/** Uploaded photo, then LinkedIn, then GitHub. */
export function effectiveContactPhotoUrl(
  profile?: Pick<ContactProfile, "photoUrl" | "linkedinPhotoUrl" | "githubPhotoUrl" | "effectivePhotoUrl">
): string | undefined {
  if (!profile) return undefined;
  const fromApi = profile.effectivePhotoUrl?.trim();
  if (fromApi) return fromApi;
  return (
    profile.photoUrl?.trim() ||
    profile.linkedinPhotoUrl?.trim() ||
    profile.githubPhotoUrl?.trim() ||
    undefined
  );
}

File: backend/internal/cv/contact_photo.go
package cv

import (
	"strings"

	"github.com/leo/ai-weekend/backend/graph/model"
)

// EffectiveContactPhotoURL returns the first non-empty photo URL in priority order:
// uploaded photoUrl, then linkedinPhotoUrl, then githubPhotoUrl.
func EffectiveContactPhotoURL(cp *model.ContactProfile) *string {
	if cp == nil {
		return nil
	}
	for _, candidate := range []*string{cp.PhotoURL, cp.LinkedinPhotoURL, cp.GithubPhotoURL} {
		if candidate == nil {
			continue
		}
		if trimmed := strings.TrimSpace(*candidate); trimmed != "" {
			return &trimmed
		}
	}
	return nil
}
